src/snowflake/connector/aio/auth/_auth.py

```diff
@@ -181,7 +183,65 @@ async def authenticate(
             "EXT_AUTHN_DUO_ALL",
             "EXT_AUTHN_DUO_PUSH_N_PASSCODE",
         ):
-            raise NotImplementedError("asyncio MFA not supported")
+            body["inFlightCtx"] = ret["data"].get("inFlightCtx")
+            body["data"]["EXT_AUTHN_DUO_METHOD"] = "push"
+            self.ret = {"message": "Timeout", "data": {}}
+
+            async def post_request_wrapper(self, url, headers, body) -> None:
+                # get the MFA response
+                self.ret = await self._rest._post_request(
+                    url,
+                    headers,
+                    body,
+                    socket_timeout=auth_instance._socket_timeout,
+                )
+
+            # send new request to wait until MFA is approved
+            try:
+                await asyncio.wait_for(
+                    post_request_wrapper(self, url, headers, json.dumps(body)),
+                    timeout=timeout,
```

Comment:
just curious, is this line correspond to https://github.com/snowflakedb/snowflake-connector-python/blob/main/src/snowflake/connector/auth/_auth.py#L313 ?

Reply:
yes